hil-test/tests/embassy_timers_executors.rs

```diff
@@ -0,0 +1,494 @@
+//! Embassy timer and executor Test
+//!
+//! Description of implemented tests:
+//! - run_test_one_shot_timg(): Tests Timg configured as OneShotTimer
+//! - run_test_periodic_timg(): Tests Timg configured as PeriodicTimer
+//! - run_test_one_shot_systimer(): Tests systimer configured as OneShotTimer
+//! - run_test_periodic_systimer(): Tests systimer configured as PeriodicTimer
+//! - run_test_periodic_oneshot_timg(): Tests Timg configured as PeriodicTimer
+//!   and then reconfigured as OneShotTimer
+//! - run_test_periodic_oneshot_systimer(): Tests systimer configured as
+//!   PeriodicTimer and then reconfigured as OneShotTimer
+//! - run_test_join_timg(): Tests Timg configured as OneShotTimer and wait on
+//!   two different timeouts via join
+//! - run_test_join_systimer(): Tests systimer configured as OneShotTimer and
+//!   wait on two different timeouts via join
+//! - run_test_interrupt_executor(): Tests InterruptExecutor and Thread
+//!   (default) executor in parallel
+//! - run_tick_test_timg(): Tests Timg configured as OneShotTimer if it fires
+//!   immediatelly in the case of the time scheduling was already in the past
+//!   (timestamp being too big)
+
+// esp32c2 is disabled currently as it fails
```

Comment:
Might be worth to document the c2 issue in https://github.com/esp-rs/esp-hal/issues/1524?

Reply:
Tbh I copied it from the Delay test 😅